### DISCUSSION TITLE
tutorials/lift-off-part3/01-feature-overview

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#44] [course feedback]
# tutorials/lift-off-part3/01-feature-overview

https://www.apollographql.com/tutorials/lift-off-part3/01-feature-overview

**Comment by CMH7** [comment]
Can't wait for another adventure

**Reply by sito-mi-37** [reply]
mi tambien

**Comment by vasily-mishanin** [comment]
Have some troubles while Setting up the Explorer
1. "Error getting collections" in "Operation Collections" 
2. Save Operation => Error "Could not fetch collections to save to"

**Reply by mabuyo** [reply]
Hey @vasily-mishanin - thanks for letting us know! We've been trying to reproduce this issue but no luck so far. Could you elaborate what action you took just before this error popped up? Were you logged in? Did you open up Sandbox at http://localhost:4000?

Any information or screenshots would be helpful! Thanks!

**Reply by jagrajbains** [reply]
Hey @vasily-mishanin, this happened with me too, turns out you need to be logged in GraphQL Explorer (localhost:4000)
In case this happens even after you are logged in the explorer...
Make sure you sign in here https://studio.apollographql.com/ and verify your email address
Once your email ID is verified you should refresh your local GraphQL explorer and you'll be able to save your collection

**Comment by misstarn** [comment]
Hello, why can't I access the api

**Reply by lizhennessy** [reply]
Hi there @misstarn  - what error are you seeing exactly? I'm happy to take a look.

**Comment by neogeek13** [comment]
"Note: This course is available in both JavaScript and TypeScript. Confirm your language of choice at the top of the lesson before continuing."

Thank you! <3

**Comment by jimischenk** [comment]
Having some sort of SSL issue:

ERROR: request to https://odyssey-lift-off-rest-api.herokuapp.com/tracks failed, reason: write EPROTO 00A1D7E601000000:error:0A000152:SSL routines:final_renegotiate:unsafe legacy renegotiation disabled:../deps/openssl/openssl/ssl/statem/extensions.c:922:

**Reply by jimischenk** [reply]
Downgrading to Node v16 fixed thsi problem

**Reply by steffanynaranjov** [reply]
adding

process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

in the server index,js after the TrackAPI will fix it

```js
const TrackAPI = require('./datasources/track-api');
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

**Comment by sarah-adams** [comment]
The server command should be 
```
npm install && npm run start
```
(start not dev, as the dev command does not exist)

**Reply by lizhennessy** [reply]
Hi @sarah-adams ! Which version of the course are you doing? You can toggle it using the dropdown at the top of the screen. For the JavaScript version, you're correct! We do use `npm start` instead. However if you're following the TypeScript version (and have checked out the `typescript` branch of the repo) there is indeed a `dev` script to launch the dev server! 

<img width="569" alt="image" src="https://github.com/apollographql-education/content-feedback/assets/95302380/dc2fd20a-0e7f-42c0-8d1f-bc1f152b5012">

I hope this helps, let us know if you have any other questions!

**Reply by sarah-adams** [reply]
Yes I am using the Typescript version.

**Reply by lizhennessy** [reply]
Ok! Great. When you open up your code and check the branch, do you see that you're on the `typescript` branch? Here's a [link](https://github.com/apollographql/odyssey-lift-off-part3/blob/typescript/server/package.json) to that repo directly that reflects what you should see in the `package.json`.

**Reply by sarah-adams** [reply]
I think I may have inadvertently cloned the javascript version of the repo instead of typescript, I re-cloned and I don't see this issue anymore. Thanks so much for the quick response!

**Reply by lizhennessy** [reply]
Oh awesome @sarah-adams so glad to hear you've got it working now. Do let us know if anything else comes up along the way, and we'll be happy to help. :)

**Comment by alx168** [comment]
Hi there, 

I'm having an issue with running the client. Does this have to do with the code generation failing? Here's the error in full with some personal information redacted:

[plugin:vite:import-analysis] Failed to resolve import "../__generated__" from "src\pages\tracks.tsx". Does the file exist?
<REDACTED>odyssey-lift-off-part3/client/src/pages/tracks.tsx:4:20
2  |  import { Layout } from "../components";
3  |  import QueryResult from "../components/query-result";
4  |  import { gql } from "../__generated__";
   |                       ^
5  |  import { useQuery } from "@apollo/client";
6  |  import TrackCard from "../containers/track-card";
    at formatError (file:///C:/Users<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:44062:46)
    at TransformContext.error (file:///C:/Users<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:44058:19)
    at normalizeUrl (file:///C:/Users/<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:41844:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///C:/Users/<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:41998:47
    at async Promise.all (index 3)
    at async TransformContext.transform (file:///C:/Users/<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:41914:13)
    at async Object.transform (file:///C:/Users/<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:44352:30)
    at async loadAndTransform (file:///C:/Users<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:55026:29)
    at async viteTransformMiddleware (file:///C:/Users/<REDACTED>/odyssey-lift-off-part3/client/node_modules/vite/dist/node/chunks/dep-41cf5ffd.js:64430:32
Click outside, press Esc key, or fix the code to dismiss.
You can also disable this overlay by setting server.hmr.overlay to false in vite.config.js.

**Reply by alx168** [reply]
I've also confirmed that I've cloned the typescript branch and am using the correct typescript tutorial. I also started up the server first before the client.

**Reply by alx168** [reply]
PS > npm run generate

> catstronauts-client@1.0.0 generate
> graphql-codegen

node:internal/modules/cjs/loader:1228
  throw err;
  ^

Error: Cannot find module './yamlAST'
Require stack:
- odyssey-lift-off-part3\client\node_modules\yaml-ast-parser\dist\src\loader.js
- odyssey-lift-off-part3\client\node_modules\yaml-ast-parser\dist\src\index.js
- odyssey-lift-off-part3\client\node_modules\@graphql-tools\prisma-loader\cjs\prisma-yml\utils\yamlComment.js
- odyssey-lift-off-part3\client\node_modules\@graphql-tools\prisma-loader\cjs\prisma-yml\PrismaDefinition.js
- odyssey-lift-off-part3\client\node_modules\@graphql-tools\prisma-loader\cjs\prisma-yml\index.js
- odyssey-lift-off-part3\client\node_modules\@graphql-tools\prisma-loader\cjs\index.js    
- odyssey-lift-off-part3\client\node_modules\@graphql-codegen\cli\cjs\graphql-config.js   
- odyssey-lift-off-part3\client\node_modules\@graphql-codegen\cli\cjs\config.js
- odyssey-lift-off-part3\client\node_modules\@graphql-codegen\cli\cjs\cli.js
- odyssey-lift-off-part3\client\node_modules\@graphql-codegen\cli\cjs\bin.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)     
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (odyssey-lift-off-part3\client\node_modules\yaml-ast-parser\dist\src\loader.js:3:11)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)      
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Module.require (node:internal/modules/cjs/loader:1311:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    odyssey-lift-off-part3\\client\\node_modules\\yaml-ast-parser\\dist\\src\\loader.js',
    odyssey-lift-off-part3\\client\\node_modules\\yaml-ast-parser\\dist\\src\\index.js',
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-tools\\prisma-loader\\cjs\\prisma-yml\\utils\\yamlComment.js', 
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-tools\\prisma-loader\\cjs\\prisma-yml\\PrismaDefinition.js',   
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-tools\\prisma-loader\\cjs\\prisma-yml\\index.js',
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-tools\\prisma-loader\\cjs\\index.js',
   odyssey-lift-off-part3\\client\\node_modules\\@graphql-codegen\\cli\\cjs\\graphql-config.js',
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-codegen\\cli\\cjs\\config.js',
    odyssey-lift-off-part3\\client\\node_modules\\@graphql-codegen\\cli\\cjs\\cli.js',
   odyssey-lift-off-part3\\client\\node_modules\\@graphql-codegen\\cli\\cjs\\bin.js'
  ]
}

Node.js v20.17.0

**Reply by alx168** [reply]
Those were the results when I ran npm run generate in the client. The same command in the server worked fine.

**Reply by alx168** [reply]
Started over with a fresh install. Turns out the problem was when I tried to update some deprecated packages with 'npm audit fix'.

**Comment by otlyn** [comment]
Fantastic tutorials—thank you for creating and sharing them! Just a heads-up: it looks like the adorable cat pictures are no longer hosted.

**Reply by lizhennessy** [reply]
thanks for letting us know, @otlyn ! These should now be fixed. :)
